src/requirements/data/majors/spanish.ts

```diff
@@ -0,0 +1,55 @@
+import { Course, CollegeOrMajorRequirement } from '../../types';
+import {
+  includesWithSubRequirements,
+  ifCodeMatch,
+  courseMeetsCreditMinimum,
+} from '../checkers-common';
+
+const spanishRequirements: readonly CollegeOrMajorRequirement[] = [
+  {
+    name: 'Core Courses',
+    description: 'Students are required to complete 4 core courses.',
+    source: 'https://romancestudies.cornell.edu/spanish-undergraduate#major-requirements',
+    checker: includesWithSubRequirements(
+      ['SPAN 2140', 'SPAN 2235'],
+      ['SPAN 2150', 'SPAN 2205'],
+      ['SPAN 2170'],
+      ['SPAN 2180']
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1, 1, 1, 1],
+    slotNames: ['SPAN 2140 or SPAN 2235', 'SPAN 2150 or SPAN 2205', 'SPAN 2170', 'SPAN 2180'],
+  },
+  {
+    name: 'Electives',
+    description:
+      'Students are required to complete 15 credits of courses focusing on the Hispanic world. SPAN 2090, SPAN 2070 and SPAN 2095 are excluded. Only courses beginning at SPAN 2130 count.',
+    source: 'https://romancestudies.cornell.edu/spanish-undergraduate#major-requirements',
+    checker: [
+      (course: Course): boolean =>
+        (ifCodeMatch(course.subject, 'SPAN') || ifCodeMatch(course.subject, 'PORT')) &&
+        !(ifCodeMatch(course.catalogNbr, '1***') || ifCodeMatch(course.catalogNbr, '20**')) &&
+        courseMeetsCreditMinimum(course, 3),
+    ],
+    fulfilledBy: 'credits',
+    perSlotMinCount: [15],
+  },
+  // TODO: see if there is a way to identify senior seminar courses.
+  {
+    name: 'Senior Seminar',
+    description: 'Must include a senior seminar in the 15 credits of electives.',
+    source: 'https://romancestudies.cornell.edu/spanish-undergraduate#major-requirements',
+    allowCourseDoubleCounting: true,
+    checker: [
+      (course: Course): boolean =>
+        (ifCodeMatch(course.subject, 'SPAN') || ifCodeMatch(course.subject, 'PORT')) &&
+        ifCodeMatch(course.catalogNbr, '4***'),
+    ],
+    checkerWarning: 'We do not check that the course is a senior seminar.',
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
```

Comment:
If the Senior Seminar is supposed to be included in the 15 credits for Electives, I believe you'll have to make this subrequirement allow double counting. `allowCourseDoubleCounting: true` should be added. I would add a comment that says we have to allow double counting here so that the Senior Seminar req can be double counted with Electives.

Reply:
Wait, I have added allowCourseDoubleCounting: true to the senior seminar requirement. Do I also need to add it for the 15 credit requirement?

Reply:
Oh sorry I missed that! Actually I think it works when adding from the Add Modal. I was trying to do this by dragging from the sidebar which is a pre-existing bug that we have to resolve. Disregard this requested change!